lifelines/statistics.py

```diff
@@ -180,23 +156,56 @@ def multivariate_logrank_test(event_durations, groups, event_observed=None,
     assert abs(Z_j.sum()) < 10e-8, "Sum is not zero."  # this should move to a test eventually.
 
     # compute covariance matrix
-    V_ = n_ij.mul(np.sqrt(d_i) / n_i, axis='index').fillna(1)
+    factor = (((n_i - d_i) / (n_i - 1)).replace(np.inf, 1)) * d_i
+    n_ij['_'] = n_i.values
+    V_ = n_ij.mul(np.sqrt(factor) / n_i, axis='index').fillna(1)
     V = -np.dot(V_.T, V_)
     ix = np.arange(n_groups)
-    V[ix, ix] = V[ix, ix] + ev
+    V[ix, ix] = -V[-1, ix] + V[ix, ix]
+    V = V[:-1, :-1]
 
     # take the first n-1 groups
-    U = Z_j.ix[:-1].dot(np.linalg.pinv(V[:-1, :-1]).dot(Z_j.ix[:-1]))  # Z.T*inv(V)*Z
+    U = Z_j.iloc[:-1].dot(np.linalg.pinv(V[:-1, :-1]).dot(Z_j.iloc[:-1]))  # Z.T*inv(V)*Z
```

Comment:
`iloc` is more defensive than `ix`. If an input duration had `-1`, this value would be in the index, and potentially fail.